### PR TITLE
Subpage Navigation via Buttons

### LINKED ISSUE #87: Enhance Navbar
- [x] suboptimal alignment of menu button and SGB logo on smaller screens (#145)
- [ ] turn dropdown buttons into [split dropdown buttons](https://getbootstrap.com/docs/4.0/components/dropdowns/#split-button-dropdowns)

### Needs to be discussed ([Wiki](https://wiki.biozentrum.unibas.ch/display/dgpsgb/2024-04-10+Workshop+Forschungsdatenplattform)):
- [x] suggestion: swap "Bibliographie" (external content) and "Daten" (internal content)
- [x] maybe add timeline, eras, ... pages to the top of browse page, 
- [x] Logo placement: portal vs. platform (Cristina)

### LINKED ISSUE #146: Remove sorting by date on the browse page
Since the compound (parent) objects do not have a date, sorting by date makes no sense.

### PR DESCRIPTION
**Pull request by mtwente**
# Pull request

## Proposed changes

* remove dropdowns from navbar, fixes #87 
  * `data.html`
    * turn *Datenarchiv* into its own button on `data.html`
    * remove separate *Quellcode* button (still available via a modal on `data.html`, could also be added to the footer if necessary).
  * `browse.html`
    * add button next to the filter bar for changing views to subjects, timeline and wordcloud. Fixes #146 
  * `timeline.html`
    * add button to the right for changing views to browse, subjects and wordcloud
    * remove separate button for selecting each year separately, since with more data, the dropdown menu is becoming longer than the Rhine). We could have the dropdown selection with centuries? But that would still be a large dropdown menu with the entries from Volume 1.
  * `subjects.html`
    * add button to the right for changing views to browse, timeline and wordcloud
  * `eras.html`
    * add button to the right for changing views to browse, timeline and subjects.
    * this button is centered on smaller screens for better visual appearance

* minor CSS improvements for `browse.html` regarding colouring and mobile view
* add "Ansicht umschalten", "Zeitachse", "Themen", "Epochen" as entries to `translations.yml` (ignored translations to Spanish...). I am not sure about the wording/translation for `browse`. Now it does not really fit to the other categories (verb vs. noun), but I did not have any good idea for phrasing at the moment.
 
## Types of changes

- [ ] New feature (non-breaking change which adds functionality).
- [x] Enhancement (non-breaking change which enhances functionality)
- [ ] Bug Fix (non-breaking change which fixes an issue).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist

- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Release Notes

- **New Features**
	- Added multilingual support with new translation keys for various UI elements: "Download," "Filtered Metadata (Excel)," "Filtered Metadata (CSV)," "More Exports," "Objects," "Timeline," "Subjects," "Eras," and "Switch View."
	- Introduced a dropdown menu for switching views in the cloud layout.

- **User Interface Enhancements**
	- Updated button labels for data export functionality to improve clarity.
	- Reorganized layout for better usability, including replacing sorting options with view-switching mechanisms.
	- Enhanced styling for input fields and interactive elements for improved user experience.
	- Updated content in the "Research Data" page to reflect new focus and improved clarity.
	- Improved layout and navigation structure on the "Browse" and "Timeline" pages for better accessibility.
	- Adjusted button layout in the timeline and list pages for a more intuitive navigation experience.
	- Implemented responsive design for button layouts based on screen size.

### CODE REVIEW

**Review by maehr — CHANGES_REQUESTED** [review]
Hi @mtwente, thanks for your PR. It solves a lot of latent problems. Here's my 5 cents:

- [ ] On `/browse.html` please move the button group to the left and change the dropdown to a btn-group-toggle (see <https://getbootstrap.com/docs/4.0/components/buttons/#checkbox-and-radio-buttons>). 
- [ ] On `/data.html` can you please add some text explaining the features (i.e. Zenodo Archive etc.)?
- [ ] Please review the feedback from coderabbitai. Especially the localisation seems to be inconsistent.

File: _data/config-nav.csv [hunk 1]
```diff
@@ -1,13 +1,5 @@
...
-Themen,/subjects.html,Sammlung,
-Zeitachse,/timeline.html,Sammlung,
-Epochen,/eras.html,Sammlung,
+Sammlung,/browse.html,,
 Bibliografie,https://www.zotero.org/groups/5004193/stadt.geschichte.basel/library,,true
-Daten,,,
-Metadaten,/data.html,Daten,
-Datenarchiv,https://zenodo.org/communities/stadt-geschichte-basel/,Daten,true
-Quellcode,https://github.com/Stadt-Geschichte-Basel/,Daten,true
-Portal,https://stadtgeschichtebasel.ch/,,true
```
Comment (maehr) [inline _data/config-nav.csv:12]: Please revert the link to the Portal

**Comment by mtwente** [comment]
> * [ ]  On `/browse.html` please move the button group to the left and change the dropdown to a btn-group-toggle (see https://getbootstrap.com/docs/4.0/components/buttons/#checkbox-and-radio-buttons).

Like this? And on the other pages, I keep it as a dropdown and on the right side?

btn-group

**Comment by maehr** [comment]
> > * [ ]   On `/browse.html` please move the button group to the left and change the dropdown to a btn-group-toggle (see https://getbootstrap.com/docs/4.0/components/buttons/#checkbox-and-radio-buttons).
> 
> Like this? And on the other pages, I keep it as a dropdown and on the right side?
> 
> btn-group

This looks great! Thank you

**Review by koilebeit — APPROVED** [review]
I like it!! It's much better than before (everything in the main nav)

But we should have some kind of dropdown button on the timeline to jump to certain points in time, otherwise you scroll forever. But that can be a separate issue/pr

**Review by maehr — CHANGES_REQUESTED** [review]
Thanks again for the changes. Here are some minor remarks:

- [x] The buttons are sized differently on `browse` vs `cloud`, `list` and `timeline_edtf`
- [ ] The button group should have the same order on all pages
- [ ] The button group should have 4 elements, i.e. links to `browse`, `cloud`, `list` and `timeline_edtf`, on all pages
- [ ] The button group should indicate which button is active (aria and visually)

Thanks again. I really like it!

**Comment by mtwente** [comment]
> - [ ] The buttons are sized differently on `browse` vs `cloud`, `list` and `timeline_edtf`

I made the buttons bigger on `browse` so it aligns with the search bar. Smaller buttons on `browse` look like they're detached from the rest of the content (as with the dropdown-button to the right in the current version of the page). On the other pages, I found large buttons to be too much.

I could change the search bar from large to small.

> - [ ] The button group should have 4 elements, i.e. links to `browse`, `cloud`, `list` and `timeline_edtf`, on all pages
> - [ ] The button group should have the same order on all pages

They do have the same order (Browse/Timeline/List/Eras), it's just that the active page/view is not shown as button (heading only). If I put all four buttons next to each other on all pages, it takes away quite a lot of space (especially on `browse`, there is very little space left for the search bar). That's also why I removed the button for the active view.

I could try moving the four buttons on their own row, stretch them across the whole page and have the search box below as-is, but that just creates oversized buttons.

**Comment by mtwente** [comment]
Here are a few visual modifications:

Medium-sized buttons, to the left.
This way, the btn-group is vertically aligned with the filter box, but not with the btn-groups on the other pages.
btn-md-left

Large-sized buttons, to the left.
btn-lg-left

Medium-sized buttons, stretched on top.
btn-md-stretch

File: _layouts/browse.html [hunk 2]
```diff
@@ -5,73 +5,87 @@
...
+        role="group"
+        aria-label="Anzeigeoptionen"
+      >
+        <a
+          href="{{ '/browse.html' | relative_url }}"
+          class="btn btn-primary btn-lg active"
+          aria-pressed="true"
+          aria-label="Zeige die Sammlungsübersicht"
+        >
+          {{ site.data.translations._layouts['browse.html'].objects[site.lang] | default: 'Objects' }}
```
[the remark this answers is not in the document]
Comment (mtwente) [reply _layouts/browse.html:23]: Translations for `objects`, `timeline`, `subjects` and `eras` are available in `_data/translations.yml` in lines 601 to 615.

**Review by maehr — APPROVED** [review]
@mtwente Thanks again for the improvement. On mobile we still run into problems (I also tested it on my native device)

savage-poltergeist-g4r44g9gr6p2w4gg-4000 app github dev_browse html(iPhone 12 Pro)

**Comment by mtwente** [comment]
> @mtwente Thanks again for the improvement. On mobile we still run into problems (I also tested it on my native device)

Yes, I was just trying a few different solutions for that!

File: _includes/js/table-js.html [hunk 3]
```diff
@@ -23,14 +23,13 @@
           "paging": true,
           "lengthMenu": [[ 25, 50, 100, -1], [ 25, 50, 100, "∞"]],
           // add download features
-          dom: 'B<"row mt-2"<"col-md-6"l><"col-md-6"f>>t<"row"<"col-md-6"i><"col-md-6"p>>',
+          dom: 'B<"d-flex justify-content-between align-items-center flex-nowrap mt-2"<"pagination-controls"l><"search-box"f>>t<"row"<"col-md-6"i><"col-md-6"p>>',
           buttons: [
               {
                   extend: 'collection',
-                  text: 'Tabelle exportieren',
+                  text: 'Herunterladen',
```
Comment (maehr) [inline _includes/js/table-js.html:30]: @mtwente localization?

**Review by maehr — APPROVED** [review]
As discussed over the phone: Great PR, fix the localization either in this or in separate PR. If separate PR please open an issue.
cheers moritz

**Pull request merged**